lib/adyen/api/recurring_service.rb

```diff
@@ -164,6 +165,21 @@ def parse_bank_details(bank)
             :holder_name      => bank.text('./payment:ownerName')
           }
         end
+
+        def parse_additional_data(xpath)
+          if xpath.empty?
+            {}
+          else
+            results = {}
+            xpath.map do |node|
+              key = node.text('./recurring:entry/recurring:key')
+              value = node.text('./recurring:entry/recurring:value')
+              results[key] = value unless key.empty?
```

Comment:
What is the case where `key` is empty?

Reply:
Good point, I'm not actually sure if there are cases where it could be empty, but I took the implementation from elsewhere, so I'm a little more reluctant to remove it.